src/odin/spare/spare_job.py

```diff
@@ -0,0 +1,383 @@
+"""
+Process data from the Spare API into parquet files.
+
+Spare API JSON responses are polled and written into an incoming bucket by Delta.
+This job processes those JSON responses into tabular data, and writes them into a parquet file.
+That output file is located at {target_location}/{endpoint_name}.parquet
+
+There is one job instance per Spare API endpoint.
+So far, only one endpoint (vehicles) is supported. More will be added later.
+Each time the job runs, it processes all files that Delta has created for the endpoint in a batch.
+
+Note: Vehicle data changes rarely, and this file is optimized for that one case.
+As other endpoints with other needs are added, different logic will be added.
+
+Input files must have a timestamp in them.
+
+The output parquet file is logically append-only.
+Changes and deletions are appended to the end of the parquet file.
+Unchanged data is ignored, so that the parquet file does not grow unnecessarily large.
+
+Changes are based on the `id` field that Spare includes for every item in its API.
```

Comment:
Based on this explanation, I would suggest modifying the structure of your parquet file. 

Generally columns created by odin, that do not exist in the original dataset are prefixed with `odin_` to avoid confusion. 

In this instance I would suggest adding an `odin_timestamp` column, equivalent to your existing `ROW_TIMESTAMP`. And instead of `ROW_DELETED` something like `odin_most_recent` as flag field containing either `NULL` or `Y` (or any equivalent flag distinction). 

With this structure, anything flagged as `odin_most_recent` was provided in the last API call and all other records are considered history. This would make pulling the latest data (for update operations) consistently performant as only one filtering action would be required.